lib/puppetserver/ca/utils/http_client.rb

```diff
@@ -136,10 +136,15 @@ def delete(url_override = nil, headers = {})
         # Like URI, but not... maybe of suspicious value
         URL = Struct.new(:protocol, :host, :port,
                          :endpoint, :version,
-                         :resource_type, :resource_name) do
+                         :resource_type, :resource_name, :queried_state) do
                 def full_url
-                  protocol + '://' + host + ':' + port + '/' +
-                  [endpoint, version, resource_type, resource_name].join('/')
+                  if queried_state
+                    return protocol + '://' + host + ':' + port + '/' +
+                         [endpoint, version, resource_type, resource_name].join('/') + "?state=" + queried_state
```

Comment:
Is the `state` parameter something that all endpoints accept?

If not I don't think we should be hardcoding like this. I know of other http apis that let you pass a Hash of query parameters and then convert that into the appropriate string. Something like that might work if `state` isn't a universal parameter to the CA.

Reply:
I took inspiration from this `https://localhost:8140/puppet-ca/v1/certificate_statuses/fake?state=signed` since it was how I use the certificate status endpoint.

Reply:
I think @justinstoller is right - this is a generic URL constructor, so probably shouldn't have `state` hardcoded. I think it would make more sense to accept arbitrary query params.